### PR TITLE
IoT Atlas - Software Update Implementation

### PR DESCRIPTION
**Pull request by court-samp**
Issue #, if available:

Description of changes: Addition of IoT Atlas Implementation - Software Updates using AWS IoT Jobs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gadams999 — CHANGES_REQUESTED** [review]
can you revert any changes to the `make_hugo.sh` script? We cannot start a validation process when modifying that file.

recent changes provide the ability to inject a `GOPROXY` setting as needed, and not hardcode to direct.

**Comment by gadams999** [comment]
Hi! thanks for the submission. I've commented on the PR to discard changes to the make_hugo.sh file. Also, as this is a new section, can you review and include the `_index.md` file that will be in the _Software Update_ section? You can see the structure and use one from one of the other implementation sections such as _Command_.

**Comment by court-samp** [comment]
revertd changes to the make_hugo.sh file, added _index.md file

**Comment by gadams999** [comment]
Hi Courtney,

I'll review this tomorrow. Working a priority review of some other repositories at this time (log4j stuff).

File: src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+## Assumptions
+This implementation approach assumes devices are already provisioned within AWS IoT. Please refer to the [Getting started with AWS IoT Core](https://docs.aws.amazon.com/iot/latest/developerguide/iot-gs.html) for details on connecting your device to AWS IoT. 
+This implementation also assumes that proper permissions have already been granted within IoT Core Policies and S3 resources. 
+
+## Implementation
+
+To create a job, you must first create a job document that is a description of the remote operations to be performed by the devices. A job document contains one or more URLs where the device can download an update or some other data. The job document can be stored in an Amazon S3 bucket, or be included inline with the command that creates the job.
+
+__Sample Job Document__:
+{{< tabs groupId="device-code">}}
```
Comment (gadams999) [inline src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md:39]: With the job document and python code and separate tabs in the same group, only one can be expanded at a time. I'd suggest getting rid of the job document as a tab, unless you plan on additional examples. If that's the case, select unique `groupId` entries for the documents and code.

File: src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+{{% notice note %}}
+For more job document examples, see the [jobs-agent.js](https://www.npmjs.com/package/aws-iot-device-sdk) example in the AWS IoT SDK for JavaScript.
+{{% /notice %}}
+
+{{< tabs groupId="device-code">}}
+{{% tab name="python" %}}
+
+Please refer to this [jobs sample](https://github.com/aws/aws-iot-device-sdk-python-v2/blob/main/samples/jobs.py) for more details and examples.
+
+- Install SDK from PyPI: `python3 -m pip install AWSIoTPythonSDK`
```
Comment (gadams999) [inline src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md:69]: The v1 device SDKs have been deprecated. This code can be left in but I'll open an issue to have it refactored using the v2 device SDK.

File: src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+{{% notice note %}}
+Your job document can contain a presigned Amazon S3 URL that points to your code file (or other file). Presigned Amazon S3 URLs are valid for a limited amount of time and so are not generated until a device requests a job document. Because the presigned URL has not been created when you are creating the job document, you put a placeholder URL in your job document instead. A placeholder URL looks like the following: **${aws:iot:s3-presigned-url:https://s3.region.amazonaws.com/<_bucket_>/<_codefile_>}** where bucket is the Amazon S3 bucket that contains the code file and code file is the Amazon S3 key of the code file
+{{% /notice %}}
+
+{{% notice note %}}
+For more job document examples, see the [jobs-agent.js](https://www.npmjs.com/package/aws-iot-device-sdk) example in the AWS IoT SDK for JavaScript.
+{{% /notice %}}
+
+{{< tabs groupId="device-code">}}
```
Comment (gadams999) [inline src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md:64]: See above to selecting a unique `groupId`

**Review by gadams999 — CHANGES_REQUESTED** [review]
Overall content, grammar, and layout are good. A couple small changes for horizontal tabs, then we can run this through!

File: src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md (hunk 1 above)
Comment (court-samp) [reply src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md:39]: removed tab for job document

File: src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md (hunk 3 above)
Comment (court-samp) [reply src/hugo/content/en-us/implementations/aws/software_update/IoT_Jobs/_index.md:64]: removed job document tab above

**Pull request merged**